tds-dll-mysql/src/main/java/tds/dll/mysql/ItemSelectionDLL.java

```diff
@@ -154,92 +163,101 @@ public SingleDataResultSet AA_GetNextItemCandidates_SP(
+					// isSegmentSatisfied (oppkey, segment) will be implemented
+					validSegments.addRecords(
+							ValidateAndReturnSegmentData(algorithm, isSim, connection, oppkey, language
+									, segment, segmentKey, segmentID, ftcnt, starttime, session));
+				} else {// segment is satisfied, mark and continue loop, NOTE:
 					// T_InsertItems
 					// modified to do this update, so this should never be
 					// necessary
 
-				final String SQL_QUERY4 = "update testopportunitysegment set IsSatisfied = 1 where _fk_TestOpportunity = ${oppkey} and SegmentPosition = ${SegmentPosition}";
-				SqlParametersMaps parameters4 = new SqlParametersMaps().put(
-						"oppkey", oppkey).put("SegmentPosition", segment);
-				int updateCnt = executeStatement(connection, SQL_QUERY4,
-						parameters4, false).getUpdateCount();
+					final String SQL_QUERY4 = "update testopportunitysegment set IsSatisfied = 1 where _fk_TestOpportunity = ${oppkey} and SegmentPosition = ${SegmentPosition}";
+					SqlParametersMaps parameters4 = new SqlParametersMaps().put(
+							"oppkey", oppkey).put("SegmentPosition", segment);
+					int updateCnt = executeStatement(connection, SQL_QUERY4,
+							parameters4, false).getUpdateCount();
 
-				_logger.debug("Update count after Insert query in AA_GetNextItemCandidates_SP is "
-						+ updateCnt);
+					_logger.debug("Update count after Insert query in AA_GetNextItemCandidates_SP is "
+							+ updateCnt);
+				}
 			}
+			return validSegments;
 		}
+	}
 
-		_Ref<String> groupIDValue = new _Ref<String>();
-		_Ref<String> blockIDValue = new _Ref<String>();
+		private List<CaseInsensitiveMap<Object>> ValidateAndReturnSegmentData(String algorithm, boolean isSim,
+			  		SQLConnection connection, UUID oppkey, String language, Integer segment, String segmentKey,
+					String segmentID, int ftcnt, Date starttime, UUID session)
+                                                              throws ReturnStatusException {
+			_Ref<String> groupIDValue = new _Ref<>();
+			_Ref<String> blockIDValue = new _Ref<>();
 
-		if (DbComparator.isEqual(algorithm, FIXEDFORM)) {
-			_AA_NextFixedformGroup_SP(connection, oppkey, segmentKey, language,
-					groupIDValue, blockIDValue);
-		}
-		if (DbComparator.containsIgnoreCase(algorithm, ADAPTIVE)
-				&& DbComparator.greaterThan(ftcnt, 0))
-		// the groupkey includes the blockID. We need this to ensure that we
-		// get the items from the desired block
-		{
-			_AA_NextFieldtestGroup_SP(connection, oppkey, segment, segmentKey,
-					segmentID, language, groupIDValue, blockIDValue, true);
+			if (DbComparator.isEqual(algorithm, FIXEDFORM)) {
+				_AA_NextFixedformGroup_SP(connection, oppkey, segmentKey, language,
+						groupIDValue, blockIDValue);
+			}
+			if (DbComparator.containsIgnoreCase(algorithm, ADAPTIVE)
+					&& DbComparator.greaterThan(ftcnt, 0))
+			// the groupkey includes the blockID. We need this to ensure that we
+			// get the items from the desired block
+			{
```

Comment:
Please put the `{` on the same line as the conditional.  The comments can either go above or inside the conditional.  

Same throughout the code where you have made changes unless all of this is just formatting.  The pattern in most Java applications I've worked on is the have the `{` always be on the same line of the conditional or loop.